src/Digdir.Domain.Dialogporten.Application/Common/Extensions/ClaimsPrincipalExtensions.cs

```diff
@@ -58,7 +73,64 @@ public static bool TryGetPid(this Claim? pidClaim, [NotNullWhen(true)] out strin
         return pid is not null;
     }
 
-    public static bool TryGetOrgNumber(this Claim? consumerClaim, [NotNullWhen(true)] out string? orgNumber)
+    // https://docs.altinn.studio/authentication/systemauthentication/
+    private class SystemUserAuthorizationDetails
+    {
+        [JsonPropertyName("type")]
+        public string? Type { get; set; }
+
+        [JsonPropertyName("systemuser_id")]
+        public string[]? SystemUserIds { get; set; }
+    }
+
+    public static bool TryGetSystemUserId(this ClaimsPrincipal claimsPrincipal, [NotNullWhen(true)] out string? systemUserId)
+    {
+        systemUserId = null;
+
+        if (!claimsPrincipal.TryGetClaimValue(AltinnAutorizationDetailsClaim, out var authDeailsJson))
+        {
+            return false;
+        }
+
+        var systemUserAuthDetails = JsonSerializer.Deserialize<SystemUserAuthorizationDetails[]>(authDeailsJson);
```

Comment:
For some reason, this always failed for me as `authDeailsJson ` was `{"type":"urn:altinn:systemuser" ....}` and not `[{"type":"urn:altinn:systemuser" ....}]` when using the token generator in the Postman workspace. I don't understand why this is, as the token made from the generator is clearly having an array here. In order to test the rest of the PR, I ended up adding code to wrap the `authDeailsJson` value in square brackets. 

The Postman workspace has been updated with system user support, see the pre-request script set in the "End User"-folder. The token generated is visible in the Postman console.

Reply:
If we have a parsing error, then something is horribly wrong, and we need to be made aware that we have a bug. Wrapping it in a try/catch will hide that, unless we also pull in `ILogger` and log something with high/critical severity that the Slack bot will report to us the same way as exceptions. For now, I think it would be best to just let this crash (as long as we have a check for that array-weirdness on the JSON string first).